ic-stable-structures/src/storage.rs

```diff
@@ -0,0 +1,41 @@
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+use ic_exports::candid::Principal;
+use ic_exports::ic_kit::ic;
+use ic_exports::stable_structures::memory_manager::{self, MemoryId, VirtualMemory};
+use ic_exports::stable_structures::DefaultMemoryImpl;
+
+pub use structures::{StableBTreeMap, StableCell};
+
+pub mod error;
+pub mod structures;
+
+pub type Memory = VirtualMemory<DefaultMemoryImpl>;
+
+type MemoryManager = memory_manager::MemoryManager<DefaultMemoryImpl>;
+
+#[derive(Default)]
+struct Manager(HashMap<Principal, MemoryManager>);
+
+impl Manager {
+    pub fn get(&mut self, memory_id: MemoryId) -> Memory {
+        let canister_id = ic::id();
```

Comment:
I was confused by this use of `ic::id()` because, as it returns always the same values, the `HashMap<Principal, MemoryManager>` is useless.
@F3kilo explained to me that we do it because it simplifies mocking during unit testing. 
Nevertheless, by doing this we pay the cost of using the HashMap in production which is something we should avoid. I don't know enough about the testing configuration to propose a solution (maybe a Manager trait with different implementations for production and testing?), I simply leave the comment here for further discussion.

Reply:
I added a subtask for that:
https://infinityswap.atlassian.net/browse/CPROD-1231?atlOrigin=eyJpIjoiM2JhMTJmYzFmOWI5NGQ3YzgyNzU2MGM0MTlmMGFlODQiLCJwIjoiaiJ9